packages/@react-spectrum/dialog/src/DialogTrigger.tsx

```diff
@@ -144,13 +144,48 @@ export {_DialogTrigger as DialogTrigger};
 
 function PopoverTrigger({state, targetRef, trigger, content, hideArrow, isKeyboardDismissDisabled, ...props}) {
   let triggerRef = useRef<HTMLElement>();
-
   let overlayRef = useRef<DOMRefValue<HTMLDivElement>>();
+  let isCloserPadding = false;
+
+  // For popover triggers close to the edge, decrease the container padding allowing popovers to
+  // move closer to their trigger.
+  if (triggerRef?.current) {
+    // The popover ref is not defined here, we don't know the arrow size to factor into the below.
+    let triggerRect = triggerRef.current.getBoundingClientRect();
+    let documentRect = document.body.getBoundingClientRect();
+    // 'bottom' is the default when no value is provided
+    let [axisPlacement] = props.placement?.split(' ') || ['bottom']; // Flip and RTL don't matter
+
+    // The following is the math that gets the desired beavhior in both scales, but isn't as logical
```

Comment:
```suggestion
    // The following is the math that gets the desired behavior in both scales but isn't as logical
```
what is the desired behavior? I think this goes along with my question about the multiplying by 2